src/app/services/favourites.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable, inject } from '@angular/core';
+import { Firestore, doc, getDoc, setDoc, arrayUnion, arrayRemove, updateDoc, DocumentReference, DocumentData } from '@angular/fire/firestore';
+import { User } from '@angular/fire/auth';
+import { AuthenticationService } from './authentication.service';
+import { Activity } from '../model/activity';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FavouritesService {
+  private firestore: Firestore = inject(Firestore);
+  private currentUser: User | null;
+  private userDocRef: DocumentReference<DocumentData>;
+  private userFavoritesCache: string[];
+
+  constructor(private authService: AuthenticationService) {
+    this.setupAuthStatusListener();
+    this.loadFavouritesInCache();
```

Comment:
We need to have user's auth status before loading his favorites, right? Do we need to await for auth?

Reply:
Loaded after auth obtained